### PR TITLE
Add plugin logger

### PR DESCRIPTION
**Pull request by pawamoy**
As discussed in https://github.com/mkdocs/mkdocs/discussions/3241.
Not sure if tests are needed, the code is quite trivial.

### CODE REVIEW

File: mkdocs/plugins.py [hunk 1]
```diff
@@ -510,3 +510,50 @@ def run_event(self, name: str, item=None, **kwargs):
...
+        name: The name to use with `logging.getLogger`.
+
+    Returns:
+        A logger configured to work well in MkDocs,
+            prefixing each message with the plugin package name.
+
+    Examples:
+        >>> from mkdocs.plugins import get_plugin_logger
+        >>> logger = get_plugin_logger(__name__)
+        >>> logger.info("My plugin message")
```
Comment (oprypin) [inline mkdocs/plugins.py]: Hm MkDocs never uses `>>>` in examples, I'd prefer to keep it that way

File: mkdocs/plugins.py [hunk 2]
```diff
@@ -510,3 +510,50 @@ def run_event(self, name: str, item=None, **kwargs):
             if result is not None:
                 item = result
         return item
+
+
+class _PrefixLogger(logging.LoggerAdapter):
```
Comment (oprypin) [inline mkdocs/plugins.py]: Could you rename `_PrefixLogger` to `PrefixedLogger` (also no underscore)

File: mkdocs/plugins.py (hunk 1 above)
Comment (pawamoy) [reply mkdocs/plugins.py:556]: Python code block then?

Comment (oprypin) [reply mkdocs/plugins.py]: Yes

**Review by oprypin — APPROVED** [review]
Thanks

**Comment by pawamoy** [comment]
Docs screenshot:
pluginlogger

**Pull request merged**